File: assessment_7_6_2019/second_ques.cpp
#include<iostream>

using namespace std ;

string merge_remaining_str(string str, int iterator) {
    int len = str.size() ;
    string merge = "" ;
    while(iterator < len) {
        merge+=str[iterator] ;
        iterator++ ;
    }
    return merge ;
}

string interleaving_of_string(string str1 , string str2) {
    int len1 = str1.size() , len2 = str2.size() ;
    int iterator1 = 0 , iterator2 = 0 ;
    string interleaved_str ;

    while(iterator1 < len1 && iterator2 < len2) {
        interleaved_str += str1[iterator1] ;
        interleaved_str += str2[iterator2] ;
        iterator1++ ;
        iterator2++ ;
    }

    interleaved_str += merge_remaining_str(str1 , iterator1) ;
    interleaved_str += merge_remaining_str(str2, iterator2) ;

    return interleaved_str ;
}

int main() {

    string str1 , str2 ;
    cin >> str1 >> str2 ;

    string interleaved_str = interleaving_of_string(str1 , str2) ;
    cout << interleaved_str << "\n" ;
    return 0 ;
}

File: assessment_7_6_2019/second_ques.js
const rl = require("readline")
const obj = {
    input : process.stdin ,
    output : process.stdout
} ;
const input = rl.createInterface(obj) ;
input.question(" " ,function(str){
    var str1 = str ;
    input.question(" ", function(str) {
        var str2= str ;
        let len1 = str1.length;
        let len2 = str.length ;
        let index1 = 0 , index2 = 0 ;
        str = "" ;
        while(index1 < len1 && index2 < len2) {
            str+=(str1[index1++] + str2[index2++]) ;
        }
        str+=(str1.slice(index1 ,len1)) ;
        str+=(str2.slice(index2 , len2)) ;
        console.log(str) ;
    })
});
